test/autotest.py

```diff
@@ -506,6 +506,8 @@ def testKill():
   def testCheckpoint():
     #start checkpoint
     coordinatorCmd(CKPT_CMD)
+    #wait and see if process will die shortly after checkpoint
+    sleep(S*SLOW)
```

Comment:
I should have been clearer. What I meant was that we don't touch testCheckpoint()'s definition at all. The CHECK from my pseudocode was a new one.

The pseudocode that you pasted works as well. However, purely from an aesthetic point of view, I would prefer sleep/check outside the testCheckpoint() routine since this sleep/check are not there to test checkpoint, but to test whether the computation runs normally after a successful resume.